File: code/python/2486.py
#https://leetcode.com/problems/append-characters-to-string-to-make-subsequence/
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        lenS, lenT = len(s), len(t)
        idxS = idxT = 0
        while idxS < lenS and idxT < lenT:
            if s[idxS] == t[idxT]:
                idxT += 1
            idxS += 1
        return lenT - idxT

File: code/cpp/2486.cpp
https://leetcode.com/problems/append-characters-to-string-to-make-subsequence/
class Solution {
public:
    int appendCharacters(string s, string t) {
        size_t nS = s.size(), nT = t.size(),
            idxS = 0, idxT = 0;
        while (idxS < nS and idxT < nT) {
            if (s[idxS] == t[idxT])
                idxT++;
            idxS++;
        }
        return nT - idxT;
    }
};
